[ci] python-package "regular" CI jobs failing to install dependencies

## Description

The `TASK=regular` jobs in this project's CI run the Python package's unit tests and test a few other things:

* the example notebookes in the `examples/` directory
* that `lightgbm` can be imported after all of its dependencies have been removed

Those jobs are currently failing across all operating systems, with an error like this:

```text
[NbConvertApp] Converting notebook interactive_plot_example.ipynb to notebook
[IPKernelApp] ERROR | KeyboardInterrupt caught in kernel.
[NbConvertApp] Writing 178679 bytes to interactive_plot_example.ipynb

PackagesNotFoundError: The following packages are missing from the target environment:
  - dask

Error: Process completed with exit code 255.
```

## Reproducible example

Logs from latest commit on `master`:

* Azure DevOps `Linux_latest regular` ([link](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=15695&view=logs&j=275189f9-c769-596a-7ef9-49fb48a9ab70&t=3a9e7a4a-04e6-52a0-67ea-6e8f6cfda74f))
* Azure DevOps `macOS regular` ([link](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=15695&view=logs&j=31d31d6b-9897-5af8-16dd-840b6b6b4fac&t=83900217-b42d-5232-5e45-a25e1385a5c6))
* GitHub Actions `regular (macOS-latest, Python 3.9)` ([link](https://github.com/microsoft/LightGBM/actions/runs/7551480124/job/20558779338))

## Environment info

LightGBM version or commit hash: https://github.com/microsoft/LightGBM/commit/255c93b551c13b5d9b0bd8eff83ef828b14b9547

## Additional Comments

I don't see any of the dependencies like `dask`, `dask-core`, etc. getting installed in the logs.

It seems that this might be silently failing:

https://github.com/microsoft/LightGBM/blob/255c93b551c13b5d9b0bd8eff83ef828b14b9547/.ci/test.sh#L128-L135

--- a/.ci/test.sh
+++ b/.ci/test.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set +x
+
 if [[ $OS_NAME == "macos" ]] && [[ $COMPILER == "gcc" ]]; then
     export CXX=g++-11
     export CC=gcc-11
@@ -131,7 +133,7 @@ if [[ $PYTHON_VERSION == "3.7" ]]; then
 fi
 
 # including python=version[build=*cpython] to ensure that conda doesn't fall back to pypy
-mamba create -q -y -n $CONDA_ENV \
+mamba create -v -y -n $CONDA_ENV \
     ${CONSTRAINED_DEPENDENCIES} \
     cffi \
     cloudpickle \

--- a/.ci/test.sh
+++ b/.ci/test.sh
@@ -321,7 +321,7 @@ matplotlib.use\(\"Agg\"\)\
     # importing the library should succeed even if all optional dependencies are not present
     mamba uninstall -n $CONDA_ENV --force --yes \
         cffi \
-        dask \
+        dask-core \
         distributed \
         joblib \
         matplotlib \

--- a/.ci/test.sh
+++ b/.ci/test.sh
@@ -319,7 +319,7 @@ matplotlib.use\(\"Agg\"\)\
     jupyter nbconvert --ExecutePreprocessor.timeout=180 --to notebook --execute --inplace *.ipynb || exit -1  # run all notebooks
 
     # importing the library should succeed even if all optional dependencies are not present
-    mamba uninstall -n $CONDA_ENV --force --yes \
+    conda uninstall -n $CONDA_ENV --force --yes \
         cffi \
         dask-core \
         distributed \

--- a/.ci/test.sh
+++ b/.ci/test.sh
@@ -1,7 +1,5 @@
 #!/bin/bash
 
-set +x
-
 if [[ $OS_NAME == "macos" ]] && [[ $COMPILER == "gcc" ]]; then
     export CXX=g++-11
     export CC=gcc-11
@@ -133,7 +131,7 @@ if [[ $PYTHON_VERSION == "3.7" ]]; then
 fi
 
 # including python=version[build=*cpython] to ensure that conda doesn't fall back to pypy
-mamba create -v -y -n $CONDA_ENV \
+mamba create -q -y -n $CONDA_ENV \
     ${CONSTRAINED_DEPENDENCIES} \
     cffi \
     cloudpickle \
@@ -321,7 +319,7 @@ matplotlib.use\(\"Agg\"\)\
     jupyter nbconvert --ExecutePreprocessor.timeout=180 --to notebook --execute --inplace *.ipynb || exit -1  # run all notebooks
 
     # importing the library should succeed even if all optional dependencies are not present
-    conda uninstall --force --yes \
+    mamba uninstall -n $CONDA_ENV --force --yes \
         cffi \
         dask \
         distributed \

